pilot/pkg/features/xds.go

```diff
@@ -51,14 +51,6 @@ var (
 		"UseRemoteAddress sets useRemoteAddress to true for sidecar outbound listeners.",
 	).Get()
 
-	EnableHeadlessService = env.Register(
-		"PILOT_ENABLE_HEADLESS_SERVICE_POD_LISTENERS",
-		true,
```

Comment:
I vaguely remember this flag is to let people turn down to reduce the number of listeners.

Reply:
Yes. If a stateful set has too many replicas, this will help to reduce the listeners. It would be nice to have this service specific, instead of at a global level even to achieve that goal.

Reply:
It was originally added _in case_ people didn't like the change - I haven't heard anyone actually use it. If set to false the behavior is incorrect.

There is some ongoing work to optimize this case as well which hopefully will make some progress